apis/temperature/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+#![no_std]
+
+use core::cell::Cell;
+use libtock_platform::{
+    share, subscribe::OneId, DefaultConfig, ErrorCode, Subscribe, Syscalls, Upcall,
+};
+
+pub struct Temperature<S: Syscalls>(S);
+
+impl<S: Syscalls> Temperature<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// initiate a temperature measurement used both for syncronous and asyncronous readings
+    pub fn read_temperature() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, READ_TEMP, 0, 0).to_result()
+    }
+
+    /// Register an events listener
+    pub fn register_listener<'share, F: Fn(i32)>(
+        listener: &'share TemperatureListener<F>,
+        subscribe: share::Handle<Subscribe<'share, S, DRIVER_NUM, 0>>,
+    ) -> Result<(), ErrorCode> {
+        S::subscribe::<_, _, DefaultConfig, DRIVER_NUM, 0>(subscribe, listener)
+    }
+
+    /// Unregister the events listener
+    pub fn unregister_listener() {
+        S::unsubscribe(DRIVER_NUM, 0)
+    }
+
+    /// initiate a syncronous temperature mesurement
```

Comment:
```suggestion
    /// Initiate a syncronous temperature mesurement.
```

Reply:
Please mention the measuring unit that the temperature is returned in (hundreds of centigrades).